code/game/machinery/baseball_machine.dm

```diff
@@ -0,0 +1,82 @@
+/obj/machinery/baseball_machine
+	name = "baseball pitching machine"
+	desc = "Hey batter batter!"
+	icon = 'icons/obj/turrets.dmi'
+	icon_state = "turretCover"
+	anchored = TRUE
+	density = TRUE
+	use_power = IDLE_POWER_USE				//this turret uses and requires power
+	idle_power_usage = 50		//when inactive, this turret takes up constant 50 Equipment power
+	active_power_usage = 300	//when active, this turret takes up constant 300 Equipment power
+	power_channel = EQUIP	//drains power from the EQUIPMENT channel
+	var/baseball_count = 15 // starts with 15 baseballs so we don't just spawn INFINITE amounts of baseballs
```

Comment:
This is gross; much better to add the baseballs to an actual list, then have them moved out of the machine and thrown--that way you're not needlessly creating/destroying objects.